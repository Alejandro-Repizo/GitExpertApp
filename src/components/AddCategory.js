import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInput = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim().length) {
			document.title = `GifExpertApp - ${inputValue}`;
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue('');
		}
	};

	return (
		<form className='form' onSubmit={handleSubmit}>
			<input
				type='text'
				className='form__input'
				value={inputValue}
				onChange={handleInput}
			/>

			<button className='form__button' type='submit'>
				<i className='fas fa-search'></i>
			</button>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
