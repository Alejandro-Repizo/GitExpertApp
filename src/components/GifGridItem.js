import React from 'react';

export const GifGridItem = ({ title, url }) => {
	return (
		<>
			<div className='card animate__animated animate__fadeIn'>
				<img src={url} alt={title} className='card__image' />
				<p className='card__title'>{title}</p>
			</div>
		</>
	);
};
