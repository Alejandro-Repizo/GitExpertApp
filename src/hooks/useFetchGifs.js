import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGIfs';

export const useFetchGifs = (category) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	useEffect(() => {
		getGifs(category).then((data) => setState({ data, loading: false }));
	}, [category]);

	return state;
};
