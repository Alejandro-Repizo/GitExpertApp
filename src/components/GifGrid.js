import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<div className='container'>
			<div className='wrapper wrapper--color'>
				<h3 className='container__title animate__animated animate__fadeIn'>
					<i className='fas fa-star icon icon--categories'></i>
					{category.charAt(0).toUpperCase() + category.slice(1)}
				</h3>

				{loading && (
					<h1 className='loader animate__animated animate__flash'>
						Loading...
					</h1>
				)}

				<div className='container__card'>
					{images.map((image) => (
						<GifGridItem key={image.id} {...image} />
					))}
				</div>
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
