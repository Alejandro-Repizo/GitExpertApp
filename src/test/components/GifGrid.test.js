import React from 'react';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { GifGrid } from '../../components/GifGrid';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid/>', () => {
	const category = 'Category';

	beforeEach(() => {
		useFetchGifs.mockReturnValue({ data: [], loading: true });
	});

	test('Se debe mostrar correctamente', () => {
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('El titulo debe ser igual a la categoria', () => {
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper.find('h3').text().trim()).toBe(category);
	});

	test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
		const gifs = [
			{ id: 'ABC', url: 'https://localhost', title: 'cualquier cosa' },
		];

		useFetchGifs.mockReturnValue({ data: gifs, loading: false });

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper.find('h1.loader').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
