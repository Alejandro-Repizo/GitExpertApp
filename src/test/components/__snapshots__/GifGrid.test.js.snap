// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid/> Se debe mostrar correctamente 1`] = `
<div
  className="container"
>
  <div
    className="wrapper wrapper--color"
  >
    <h3
      className="container__title animate__animated animate__fadeIn"
    >
      <i
        className="fas fa-star icon icon--categories"
      />
      Category
    </h3>
    <h1
      className="loader animate__animated animate__flash"
    >
      Loading...
    </h1>
    <div
      className="container__card"
    />
  </div>
</div>
`;
